test/base/src/androidTest/kotlin/io/realm/test/shared/RealmSchemaTests.kt

```diff
@@ -0,0 +1,183 @@
+        tmpDir = PlatformUtils.createTempDir()
+        val configuration =
+            RealmConfiguration.Builder(schema = setOf(SchemaVariations::class, Sample::class))
+                .path("$tmpDir/default.realm").build()
+        realm = Realm.open(configuration)
+    }
+
+    @AfterTest
+    fun tearDown() {
+        if (!realm.isClosed()) {
+            realm.close()
+        }
+        PlatformUtils.deleteTempDir(tmpDir)
+    }
+
+    @Test
+    fun realmClass() {
+        val schema = realm.schema()
+
+        val schemaVariationsName = "SchemaVariations"
+        val schemaVariationsDescriptor = schema[schemaVariationsName] ?: fail("Couldn't find class")
+        assertEquals(schemaVariationsName, schemaVariationsDescriptor.name)
+        assertEquals("string", schemaVariationsDescriptor.primaryKey()?.name)
+
+        val sampleName = "Sample"
+        val sampleDescriptor = schema[sampleName] ?: fail("Couldn't find class")
+        assertEquals(sampleName, sampleDescriptor.name)
+        assertNull(sampleDescriptor.primaryKey())
+    }
+
+    @Test
+    fun realmProperty() {
+        val schema = realm.schema()
+
+        val schemaVariationsName = "SchemaVariations"
+        val schemaVariationsDescriptor = schema[schemaVariationsName] ?: fail("Couldn't find class")
+
+        schemaVariationsDescriptor["string"]!!.run {
+            assertEquals("string", name)
+            type.run {
+                assertIs<SingularPropertyType>(this)
+                assertEquals(RealmStorageType.STRING, storageType)
+                assertFalse(isNullable)
+                assertTrue(isPrimaryKey)
+                assertFalse(isIndexed)
+            }
+            assertFalse(isNullable)
+        }
+        schemaVariationsDescriptor["nullableString"]!!.run {
+            assertEquals("nullableString", name)
+            type.run {
+                assertIs<SingularPropertyType>(this)
+                assertEquals(RealmStorageType.STRING, storageType)
+                assertTrue(isNullable)
+                assertFalse(isPrimaryKey)
+                assertTrue(isIndexed)
+            }
+            assertTrue(isNullable)
+        }
+        schemaVariationsDescriptor["stringList"]!!.run {
+            assertEquals("stringList", name)
+            type.run {
+                assertIs<ListPropertyType>(this)
+                assertEquals(RealmStorageType.STRING, storageType)
+                assertFalse(this.isNullable)
+            }
+            assertFalse(isNullable)
+        }
+        schemaVariationsDescriptor["nullableStringList"]!!.run {
+            assertEquals("nullableStringList", name)
+            type.run {
+                assertIs<ListPropertyType>(this)
+                assertEquals(RealmStorageType.STRING, storageType)
+                assertTrue(this.isNullable)
+            }
+            assertFalse(isNullable)
+        }
+    }
+
+    @Test
+    @Suppress("NestedBlockDepth")
+    fun schema_optionCoverage() {
```

Comment:
Maybe add a comment about this test is for ensuring that we have a class that touches all possible options for a property but _doesn't_ cover all permutations... I was a bit confused about this at least.